inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -171,7 +171,7 @@ def _rollback_and_delete_snap(self) -> None:
         a.) After reboot by SOTA, and diagnostic reports bad report for system health
         """
         logger.debug("")
-        if self.snap_num:
+        if self.snap_num and self.snap_num is not "0":
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Robustness$</h3>

Ensure that `self.snap_num` is always a string before performing string comparisons, or convert it explicitly to avoid type-related bugs.

Reply:
Yes already it is declared.